parser_library/src/workspace_manager.cpp

```diff
@@ -991,6 +990,22 @@ class workspace_manager_impl final : public workspace_manager,
     std::vector<fade_message> m_fade_messages;
     unsigned long long m_unique_id_sequence = 0;
 
+    static std::string_view extract_scheme(std::string_view uri) { return uri.substr(0, uri.find(':') + 1); }
+    void recompute_allow_list()
+    {
+        allowed_schemes.assign(std::begin(default_allowed_schemes), std::end(default_allowed_schemes));
```

Comment:
Just curious - is there a benefit of using `std::vector` instead of e.g. `std::map` or `std::set` here? Was the reason for using `std::vector` that it's enough for this use case?

Reply:
This is expected to be a small structure (<20 elements) of mostly short strings.